rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -270,6 +270,103 @@ def __init__(self,
+                 The iterator would return the first integer multiple
+                 of self._num_gpus * self.batch_size entries which exceeds 'size'.
+                 Setting this flag to False will cause the iterator to return
+                 exactly 'size' entries.
+    dynamic_shape: bool, optional, default = False
+                 Whether the shape of the output of the RALI pipeline can
+                 change during execution. If True, the pytorch tensor will be resized accordingly
+                 if the shape of RALI returned tensors changes during execution.
+                 If False, the iterator will fail in case of change.
+    last_batch_padded : bool, optional, default = False
+                 Whether the last batch provided by RALI is padded with the last sample
+                 or it just wraps up. In the conjunction with `fill_last_batch` it tells
+                 if the iterator returning last batch with data only partially filled with
+                 data from the current epoch is dropping padding samples or samples from
+                 the next epoch. If set to False next epoch will end sooner as data from
+                 it was consumed but dropped. If set to True next epoch would be the
+                 same length as the first one.
+    Example
+    -------
+    With the data set [1,2,3,4,5,6,7] and the batch size 2:
+    fill_last_batch = False, last_batch_padded = True  -> last batch = [7], next iteration will return [1, 2]
+    fill_last_batch = False, last_batch_padded = False -> last batch = [7], next iteration will return [2, 3]
+    fill_last_batch = True, last_batch_padded = True   -> last batch = [7, 7], next iteration will return [1, 2]
+    fill_last_batch = True, last_batch_padded = False  -> last batch = [7, 1], next iteration will return [2, 3]
+
+    """
+    def __init__(self, pipeline, tensor_layout = types.NONE, tensor_dtype = types.FLOAT, size = -1, auto_reset = False, device = "cpu", device_id = 0):
+        self.loader = pipeline
+        self.tensor_format = tensor_layout
+        self.device = device
+        self.device_id = device_id
+        self.output = None
+        self.iterator_length = b.getRemainingImages(self.loader._handle)
+        self.max_shape = None
+        self.batch_size = self.loader._batch_size
+        self.output_list = None
+        self.labels_size = self.batch_size
+        self.output_memory_type = self.loader._output_memory_type
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.loader.rocal_run() != 0:
+            raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.get_output_tensors()
+
+        self.output_list = []
+        for i in range(len(self.output_tensor_list)):
+            dimensions = self.output_tensor_list[i].dimensions()
+            self.num_roi_dims = len(dimensions) - 1
```

Comment:
There is an API called get_roi_dims_size. You can use, output_tensor_list[i].get_roi_dims_size() to get this num_roi_dims value

Please make use of it and check if it works as expected

Reply:
Works as expected

Reply:
Changes done